### ISSUE TITLE
cannot install xonsh 0.9.25 from pip

### ISSUE THREAD

**Issue by dtrodrigues** [xonsh/xonsh#4074]
## Expected Behavior
xonsh should install via pip
## Current Behavior
xonsh does not install because the version inside the sdist doesn't match the version of the filename (0.9.25.dev3 vs 0.9.25)
```
$ pip install xonsh==0.9.25
Collecting xonsh==0.9.25
  Downloading xonsh-0.9.25.tar.gz (1.4 MB)
     |████████████████████████████████| 1.4 MB 5.3 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
  WARNING: Requested xonsh==0.9.25 from https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60, but installing version 0.9.25.dev3
WARNING: Discarding https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60 (from https://pypi.org/simple/xonsh/) (requires-python:>=3.6). Requested xonsh==0.9.25 from https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60 has inconsistent version: filename has '0.9.25', but metadata has '0.9.25.dev3'
ERROR: Could not find a version that satisfies the requirement xonsh==0.9.25
ERROR: No matching distribution found for xonsh==0.9.25
```

**Comment by anki-code** [comment]
I can't repeat this:
```python
pip install xonsh==0.9.25
# Processing /home/pc/.cache/pip/wheels/18/8a/a1/81302f50bd4865eb6e4dca058f0c451831dc2b543f33f3193f/xonsh-0.9.25.dev3-py3-none-any.whl
# Installing collected packages: xonsh
#   Attempting uninstall: xonsh
#     Found existing installation: xonsh 0.9.25.dev3
#     Uninstalling xonsh-0.9.25.dev3:
#       Successfully uninstalled xonsh-0.9.25.dev3
# Successfully installed xonsh-0.9.25.dev3

# Prepared by xontrib-hist-format
```

**Comment by dtrodrigues** [comment]
I believe pip 20.3 started enforcing stricter name checks.

Older pip (works):
```
$ pip list
Package    Version
---------- -----------
pip        20.2.4

$ pip install xonsh==0.9.25
Collecting xonsh==0.9.25
  Downloading xonsh-0.9.25.tar.gz (1.4 MB)
     |████████████████████████████████| 1.4 MB 5.2 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
  WARNING: Requested xonsh==0.9.25 from https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60, but installing version 0.9.25.dev3
Building wheels for collected packages: xonsh
  Building wheel for xonsh (PEP 517) ... done
  Created wheel for xonsh: filename=xonsh-0.9.25.dev3-py3-none-any.whl size=979873 sha256=e1b0913cf947fcd9ec9f2c44076e68ad1d8a9202d5ae5218225c2e7b1a13385c
  Stored in directory: /Users/dustin/Library/Caches/pip/wheels/77/e8/5d/c82292337c05efc1144fa9ee9d265bbb032fa80ada662308f8
Successfully built xonsh
Installing collected packages: xonsh
  Attempting uninstall: xonsh
    Found existing installation: xonsh 0.9.25.dev3
    Uninstalling xonsh-0.9.25.dev3:
      Successfully uninstalled xonsh-0.9.25.dev3
Successfully installed xonsh-0.9.25.dev3
```

Newer pip (doesn't work):
```
$ pip list
Package    Version
---------- -----------
pip        20.3

$ pip install xonsh==0.9.25
Collecting xonsh==0.9.25
  Downloading xonsh-0.9.25.tar.gz (1.4 MB)
     |████████████████████████████████| 1.4 MB 5.0 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
  WARNING: Requested xonsh==0.9.25 from https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60, but installing version 0.9.25.dev3
ERROR: Requested xonsh==0.9.25 from https://files.pythonhosted.org/packages/7f/3d/a29fe1f9dc61502a0cdee70e4bd4f97164106975209b6feaa2e064d241fc/xonsh-0.9.25.tar.gz#sha256=20af3480d6ac1b8a48f03288796b16a35f498f047e3472c300e4ef5a26f03d60 has different version in metadata: '0.9.25.dev3'
```

**Comment by gforsyth** [comment]
Thanks for reporting @dtrodrigues -- not sure how those extra commits got folded in during release.  I'll try to bump to 0.9.26 today to fix this

**Comment by richerve** [comment]
The AppImage linked on the 0.9.25 release is also showing an incorrect version, actually pointing to 0.9.24 instead: `0.9.24.dev223`

This is the information from running `xonfig`:

```
| xonsh            | 0.9.24.dev223        |
| Git SHA          | 1f7f33c9             |
| Commit Date      | Jan 25 16:37:29 2021 |
```

**Comment by richerve** [comment]
@gforsyth is the new release coming soon?

**Comment by bestlem** [comment]
Not that installing via conda I have in 
conda list xonsh
# packages in environment at /Users/mark/miniconda3/envs/xonsh:
#
# Name                    Version                   Build  Channel
xonsh                     0.9.25           py38h10201cd_0    conda-forge

but in ${...}
'XONSH_VERSION': '0.9.24'
 '_': '/Users/mark/miniconda3/envs/xonsh/bin/xonsh',

So I think version number updateing in general did not work

**Comment by gforsyth** [comment]
Hey all -- I just released 0.9.26 and I've confirmed that the pypi tarball installs using the newest version of `pip`.

Our releaser got tripped up by an unexpected flag in a requirements file and it broke a bunch of things partway (but not enough to stop the release) hence the wackiness of the previous release.  

Should be all good now.  New conda-forge release will be available soon, just waiting on the build to finish
